sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/SchemaPruning.scala

```diff
@@ -127,7 +128,8 @@ object SchemaPruning extends SQLConfHelper {
   private def getRootFields(expr: Expression): Seq[RootField] = {
     expr match {
       case att: Attribute =>
-        RootField(StructField(att.name, att.dataType, att.nullable), derivedFromAtt = true) :: Nil
+        RootField(StructField(att.name, att.dataType, att.nullable, att.metadata),
```

Comment:
Can we add a unit test in `SchemaPruningSuite`? We can make `getRootFields` as `private[spark]` to test it.

Reply:
Good point. Let me add a test.